torch/jit/frontend.py

```diff
@@ -286,14 +287,31 @@ def build_class_def(ctx, py_def, methods, properties, self_name):
 
 
 def build_def(ctx, py_def, type_line, def_name, self_name=None):
+    global _args_and_types  # type: ignore[name-defined]
+
+    _args_and_types = False  # type: ignore[name-defined]
     body = py_def.body
     r = ctx.make_range(py_def.lineno + len(py_def.decorator_list),
                        py_def.col_offset,
                        py_def.col_offset + len("def"))
+
+    # If MonkeyType is installed, get all the consolidated traces for the arguments
+    # and the return types from type_trace_db
+
+    type_trace_db = torch.jit._script.type_trace_db
```

Comment:
Can we set up a method in `_script.py` to return this global var rather than having an external module accessing a private member?

Reply:
Yes now added.